module-2/backend/week_four.md

```diff
@@ -10,14 +10,40 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+small amnt of data stored on your local machine (usually consists of a user id), which tells the server that you have been to the site before
+
 * What’s the difference between a session and a cookie?
+sessions are encrypted cookies
+
 * What’s a flash and when do you want to use flashes?
+a small notice which allows you to notify the user, on the webpage, exactly what is/is not happening in the background - they are not permanent - you want a flash if someone creates something/updates something, etc.
+
 * Why do people say “HTTP is stateless”?
+because the sessions, and associated info, lies with the client, NOT the server
+
 * What’s authentication? Explain.
+authentication is allowing access to some/all pages of a website
+
 * What’s the difference between authentication and authorization?
+authentication is allowing access to the site, in general, authorization is dictating WHAT you are allowed to access/mess with
+
 * What’s a before filter?
+a line of code, at the top of your controller, to perform a method before any of the controller methods are performed UNLESS you specify with an except
+
 * How do we keep track of a user once they’ve logged in?
+user id/other info stored in a session
+
 * When do you want to namespace a resource? When do you want to nest a resource? What's the differences between those two approaches?
+namespace - isn't necessary to have a controller, doesn't have an id (for say, admin) - admin is a great example
+resource - has an id associated with both, controller is necessary - User is a nice one - mutliple users and varying content require an id
+
 * At a high level, what tools can you use to implement authorization? How would you use them?
+admin - mass implementation/destruction of certain features on the website
+enums - assigning a default and an admin
```

Comment:
Yup! I think also `before_actions`, and sessions are the big tools that we use for authorization.